src/modules/capabilities.json

```diff
@@ -0,0 +1,704 @@
+                        "value": {
+                            "capability": "xrn:firebolt:capability:token:platform",
+                            "role": "use",
+                            "supported": true,
+                            "available": false,
+                            "permitted": false,
+                            "granted": false,
+                            "details": ["unavailable"]
+                        }
+                    }
+                }
+            ]
+        },
+        {
+            "name": "onGranted",
+            "tags": [
+                {
+                    "name": "event"
+                }
+            ],
+            "summary": "Listens for all App permitted capabilities to become granted.",
+            "params": [
+                {
+                    "name": "role",
+                    "required": true,
+                    "schema": {
+                        "$ref": "https://meta.comcast.com/firebolt/capabilities#/definitions/Role"
+                    }
+                },
+                {
+                    "name": "capability",
+                    "required": true,
+                    "schema": {
+                        "$ref": "https://meta.comcast.com/firebolt/capabilities#/definitions/Capability"
+                    }
+                }
+            ],
+            "result": {
+                "name": "value",
+                "summary": "Provides the capability info.",
+                "schema": {
+                    "oneOf": [
+                        {
+                            "$ref": "https://meta.comcast.com/firebolt/types#/definitions/ListenResponse"
+                        },
+                        {
+                            "$ref": "https://meta.comcast.com/firebolt/capabilities#/definitions/CapabilityInfo"
+                        }
+                    ]
+                }
+            },
+            "examples": [
+                {
+                    "name": "Postal code granted",
+                    "params": [
+                        {
+                            "name": "capability",
+                            "value": "xrn:firebolt:capability:localization:postalCode"
+                        },
+                        {
+                            "name": "role",
+                            "value": "use"
+                        }
+                    ],
+                    "result": {
+                        "name": "Default Result",
+                        "value": {
+                            "capability": "xrn:firebolt:capability:localization:postalCode",
+                            "role": "use",
+                            "supported": true,
+                            "available": true,
+                            "permitted": true,
+                            "granted": true
+                        }
+                    }
+                }
+            ]
+        },
+        {
+            "name": "onRevoked",
```

Comment:
Would onRevoked be called when a grant lifetime expires?

Also wondering about single-use grant lifetimes. Does it emit onGranted and onRevoked at the same time?

Reply:
Great questions...

- `onRevoked` probably should fire when an grant expires. The user is implicity revoking it at some point in the future, so I'm okay with the name here. Is this scalable for an FEE to manage?
- For single-use grants, we'd fire `onRevoked` _after_ it's used. This might be tricky for cloud APIs, though...